### ISSUE TITLE
Flink: Support Flink 1.17

### ISSUE THREAD

**Issue by PrabhuJoseph** [delta-io/connectors#534]
Flink-1.17 is [released](https://flink.apache.org/2023/03/23/announcing-the-release-of-apache-flink-1.17/). This ticket intends to support Flink-1.17 in delta-io.

**Comment by scottsand-db** [comment]
We are planning to have the next delta-flink release 0.7.0 support flink 1.16 and SQL/ Table API.
We can then have Flink 1.17 on the 0.8.0 release

**Comment by kristoffSC** [comment]
Hi @PrabhuJoseph 
I'm wondering, have you tried running 0.6.0 with Flink 1.17 maybe?

The connector **does not ship** any Flink dependencies so if there were no changes in Flink Sink/Source API then 0.6.0 and 0.7.0 could be used with 1.17.

For example, 
**0.5.0 will work only with Flink 1.13.x**
**0.6.0 will work with Flink 1.15.x and 1.16.x** even though 0.6.0 has a "dependency" on Flink 1.15. You can still create Flink jobs using Flink 1.16 libs and 0.6.0 connector on Flink 1.16.  [1]

In other words, even though connector has FlinkVersion X in build.sbt it does not mean that It will not work with Flink X + 1 version.

I've not tried 0.6.0 on 1.17 yet.

[1] https://github.com/delta-io/connectors/issues/523

**Comment by PrabhuJoseph** [comment]
Thanks @kristoffSC and @scottsand-db .

Delta build ran fine with just changing the Flink dependency version to 1.17. We are now testing with few Flink Delta Streaming examples on Flink-1.17 cluster.

**Comment by vkorukanti** [comment]
This repo has been deprecated and the code is moved under `connectors` module in https://github.com/delta-io/delta repository. Please create the issue in repository https://github.com/delta-io/delta. See delta-io/connectors#556 for details.
